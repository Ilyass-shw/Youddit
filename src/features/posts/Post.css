.post-block {
	margin-bottom: 15px;
	display: flex;
	width: 82%;
	margin-left: auto;
	margin-right: auto;
	height: 10rem;
	overflow: hidden;
}
.image-container {
	flex-basis: 15rem;
	background-color: rgb(32, 32, 33);
	display: flex;
	align-items: center;
	min-width: 230px;
}
.content-image {
	width: 100%;
	border-radius: 2.5px;
	max-height: 100%;
}
.post-data {
	margin-left: 13px;
	min-width: 0;
	color: #a5aa9c;
	flex-basis: 80%;
}
.post-data .title {
	overflow: hidden;
	text-overflow: ellipsis;
	display: -webkit-box;
	-webkit-line-clamp: 2;
	-webkit-box-orient: vertical;

	font-size: 1.2rem;
	font-weight: 300;
	color: #ffffff;
	margin-top: 0;
	margin-bottom: 2px;
}
.post-data p {
	font-size: 0.75rem;
	margin: 0;
}
.upvotes-date-data {
	display: flex;
}
.upvotes-date-data p span {
	margin: 0 4px;
}
.subreddit {
	position: relative;
	display: flex;
	align-items: flex-end;
	padding: 12px 0;
}
.subreddit svg{
	min-width: 24px;
}
.subreddit-icon {
	height: 1.7rem;
	width: 1.7rem;
	border-radius: 50%;
}
.subreddit p {
	margin-left: 7px;
}
.subreddit-label {
	padding: 0.4rem;
	color: #ffffff;
	background-color: #5a5a5ac9;
	border-radius: 2px;
	position: absolute;
	left: 22px;
    bottom: 42px;
	opacity: 0;
	transition: opacity 0.28s;
}
.subreddit-name:hover + p {
	opacity: 1;
}
