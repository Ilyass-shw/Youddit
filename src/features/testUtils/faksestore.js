import { configureStore } from "@reduxjs/toolkit";
import { createSlice } from "@reduxjs/toolkit";

const initialState = {
	firstFilter: "relevance",
	lastFilter: "comments",
	filter: "relevance",
};
const fakeSlice = createSlice({
	name: "posts",
	initialState,
	reducers: {},
});

export default configureStore({
	reducer: {
		posts: fakeSlice.reducer,
	},
});

export const makeCustomTestStore = (name, initialState, reducers = {}) => {
	const fakeSlice = createSlice({
		name,
		initialState,
		reducers,
	});

	return configureStore({
		reducer: {
			posts: fakeSlice.reducer,
		},
	});
};
