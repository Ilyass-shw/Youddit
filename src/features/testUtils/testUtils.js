import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { Provider } from "react-redux";
import store, { makeCustomTestStore } from "./faksestore";

const TestProvider = ({ store, children }) => <Provider store={store}>{children}</Provider>;

export function render(ui, { store, ...otherOpts }) {
	return rtlRender(<TestProvider store={store}>{ui}</TestProvider>, otherOpts);
}

export function makeTestStore(name, initialState, reducers) {
	if (name && initialState) {
		const Store = makeCustomTestStore(name, initialState, reducers);
		const origDispatch = Store.dispatch;
		Store.dispatch = jest.fn(origDispatch);
		return Store;
	} else {
		const origDispatch = store.dispatch;
		store.dispatch = jest.fn(origDispatch);
		return store;
	}
}

// re-export everything
export * from "@testing-library/react";
