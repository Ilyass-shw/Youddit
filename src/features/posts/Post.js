import React from "react";
import PropTypes from "prop-types";
import "./Post.css";
import { Icon } from "@iconify/react";
import bxlReddit from "@iconify/icons-bx/bxl-reddit";
import { handleThumbnail } from "../../api/api";

const Post = ({ post }) => {
	let isDesktop = window.innerWidth > 580;
	return (
		<div className="post-block">
			<a
				alt={post.title}
				href={"https://www.reddit.com" + post.url}
				target="_blank"
				rel="noopener noreferrer"
				className="image-container"
			>
				{handleThumbnail(post.img)}
			</a>
			<div className="post-data">
				<a alt={post.title} href={"https://www.reddit.com" + post.url} target="_blank" rel="noopener noreferrer">
					<div className="post-data-container">
						{!isDesktop && (
							<div className="subreddit">
								<Icon icon={bxlReddit} width="1.5rem" height="1.5rem" />
							</div>
						)}
						<div className="mobile-post-data">
							<h4 className="title">{post.title}</h4>
							<div className="upvotes-date-data">
								<p>
									{post.subreddit}
									<span className="dot">•</span>
									{post.upvotes} upvotes<span className="dot">•</span>
									{post.date}
								</p>
							</div>
						</div>
						{isDesktop && (
							<div className="subreddit">
								<Icon icon={bxlReddit} width="1.5rem" height="1.5rem" />
								<p className="subreddit-name">{post.subreddit}</p>
								<p className="subreddit-label">{post.subreddit}</p>
							</div>
						)}
					</div>
				</a>
			</div>
		</div>
	);
};

Post.propTypes = {
	post: PropTypes.object.isRequired,
};

export default Post;
