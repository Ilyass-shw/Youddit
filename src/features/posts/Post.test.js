import React from "react";
import Post from "./Post";
import { render, makeTestStore } from "../testUtils/testUtils";

jest.mock("@iconify/react", () => {
	return { Icon: () => <p>Icon mock</p> };
});
describe("Post", () => {
	it("should render without crashing ", () => {
		const store = makeTestStore();

		render(
			<Post
				post={{
					title: "fake title 1",
					img: "fake img url 1",
					url: "fake link 1",
					upvotes: "10k",
					date: "1 days ago",
					author: "author 1",
					subreddit: "subreddit 1",
					id: "id 1",
				}}
			/>,
			{ store }
		);
	});

	it("should match the snapshot on desktop", () => {
		const store = makeTestStore();
        global.innerWidth= 1000

		const { container } = render(
			<Post
				post={{
					title: "fake title 1",
					img: "fake img url 1",
					url: "fake link 1",
					upvotes: "10k",
					date: "1 days ago",
					author: "author 1",
					subreddit: "subreddit 1",
					id: "id 1",
				}}
			/>,
			{ store }
		);

        expect(container).toMatchSnapshot()
	});

	it("should match the snapshot on devices smaller than desktop", () => {
		const store = makeTestStore();
        global.innerWidth= 579

		const { container } = render(
			<Post
				post={{
					title: "fake title 1",
					img: "fake img url 1",
					url: "fake link 1",
					upvotes: "10k",
					date: "1 days ago",
					author: "author 1",
					subreddit: "subreddit 1",
					id: "id 1",
				}}
			/>,
			{ store }
		);

        expect(container).toMatchSnapshot()
	});
});
