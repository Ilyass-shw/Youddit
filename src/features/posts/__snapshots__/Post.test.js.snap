// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should match the snapshot on desktop 1`] = `
<div>
  <div
    class="post-block"
  >
    <a
      alt="fake title 1"
      class="image-container"
      href="https://www.reddit.comfake link 1"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="post"
        class="content-image"
        src="fake img url 1"
      />
    </a>
    <div
      class="post-data"
    >
      <a
        alt="fake title 1"
        href="https://www.reddit.comfake link 1"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="post-data-container"
        >
          <div
            class="mobile-post-data"
          >
            <h4
              class="title"
            >
              fake title 1
            </h4>
            <div
              class="upvotes-date-data"
            >
              <p>
                subreddit 1
                <span
                  class="dot"
                >
                  •
                </span>
                10k
                 upvotes
                <span
                  class="dot"
                >
                  •
                </span>
                1 days ago
              </p>
            </div>
          </div>
          <div
            class="subreddit"
          >
            <p>
              Icon mock
            </p>
            <p
              class="subreddit-name"
            >
              subreddit 1
            </p>
            <p
              class="subreddit-label"
            >
              subreddit 1
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`Post should match the snapshot on devices smaller than desktop 1`] = `
<div>
  <div
    class="post-block"
  >
    <a
      alt="fake title 1"
      class="image-container"
      href="https://www.reddit.comfake link 1"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="post"
        class="content-image"
        src="fake img url 1"
      />
    </a>
    <div
      class="post-data"
    >
      <a
        alt="fake title 1"
        href="https://www.reddit.comfake link 1"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="post-data-container"
        >
          <div
            class="subreddit"
          >
            <p>
              Icon mock
            </p>
          </div>
          <div
            class="mobile-post-data"
          >
            <h4
              class="title"
            >
              fake title 1
            </h4>
            <div
              class="upvotes-date-data"
            >
              <p>
                subreddit 1
                <span
                  class="dot"
                >
                  •
                </span>
                10k
                 upvotes
                <span
                  class="dot"
                >
                  •
                </span>
                1 days ago
              </p>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
